### ISSUE TITLE
Store denormalization failing to resolve object connections

### ISSUE THREAD

**Issue by simenbrekken** [mattkrick/cashay#49]
```javascript
// List page query
const flightsQuery = `
  query (
    $fromDate: String!
    $toDate: String!
    $first: Int
  ) {
    flights(
      fromDate: $fromDate
      toDate: $toDate
      first: $first
    ) {
      id
      cursor
      status
    }
  }
`

// Detail page query
const flightQuery = `
  query (
    $id: ID
  ) {
    flight(id: $id) {
      id
      departure {
        id
        airportName
      }
      arrival {
        id
        airportName
      }
    }
  }
`

// Store contents
"entities": {
  "Flight": {
    "dy310-osl-tos-20160608": {
      "id": "dy310-osl-tos-20160608",
      "cursor": "dy310-osl-tos-20160608",
      "flightStatus": null,
      "departure": "DepartureStatus:OSL2016-06-08T07:20:00Z",
      "arrival": "ArrivalStatus:TOS2016-06-08T10:10:00Z"
    },
  },
  "DepartureStatus": {
    "OSL2016-06-08T07:20:00Z": {
      "id": "OSL2016-06-08T07:20:00Z",
      "airportName": "Oslo"
    }
  },
  "ArrivalStatus": {
    "TOS2016-06-08T10:10:00Z": {
      "id": "TOS2016-06-08T10:10:00Z",
      "airportName": "Tromsø"
    }
  }
}

Attempting to read `data.flight.arrival.airportName` returns `null`
```

**Comment by mattkrick** [comment]
this one stumps me. Can you replicate the results using the [test schema](https://github.com/mattkrick/cashay/blob/master/src/__tests__/schema.js), or share the client schema? I think i'll just have to walk the debugger and see where it fails.

**Comment by simenbrekken** [comment]
@mattkrick [Here's the schema](https://gist.github.com/simenbrekken/ec51bed6a044fad56bb7ebbb4cf811e0)

**Comment by simenbrekken** [comment]
@mattkrick In 0.6.0 this goes completely ballistic and queries the server with the same query over and over.

**Comment by mattkrick** [comment]
@simenbrekken try 0.6.1, i fixed up a small bug that's probably the culprit. 

OHHHH, i think it's because i use `:` as the delimiter.

**Comment by mattkrick** [comment]
without the repo i'm kinda flying blind (pun intended), but I think the delimite is the root cause, so we can do 2 things to make sure it doesn't happen again:

1. change the delimiter to `::`
2. split the delimited string into only 2 parts on `::`

In doing so, the only way someone could bork it is if they had `::` in one of their GraphQL type names.

**Comment by simenbrekken** [comment]
@mattkrick I changed my ID generator to use `btoa` and the problem disappeared. It was a bit unfortunate that I was using dummy ID generation but I think your solution is sound

**Comment by mattkrick** [comment]
fixed in published version 0.6.2.
please re-open if it's still giving you trouble
